File: py_src/function/convert.py
# coding: utf-8
from common.operate_json import read_json, overwrite_json


# データ置換
def convert(data, key, update_value, link, symbol):
    for d in data:
        if symbol != "" and symbol in update_value:
            value = update_value.replace(symbol, str(d[link]))
        else:
            value = update_value

        if key in d:
            d.update({key: value})

        sub = d["subDatas"]
        for d2 in sub:
            if key in d2.values():
                d2.update({"attribute": key, "value": value})
        d.update({"subDatas": sub})

    return data


def convert_property(projectPath, key, value, link, symbol):
    data = read_json(projectPath)
    data = data["blueprint"]
    data = convert(data, key, value, link, symbol)
    updateData = {"blueprint": data}
    overwrite_json(updateData, projectPath)
    return "ok"

File: src/schema/convert.ts
export type ConvertContents = {
  key: string;
  value: string;
  link: string;
  symbol: string;
};

export type Convert = ConvertContents & { [key: string]: string };

export const convertHeader: Convert = {
  key: "プロパティ名",
  value: "変更後の値",
  link: "リンクさせるプロパティ名",
  symbol: "リンクさせる独自の記号",
};

export const initConvertDatas: Convert = {
  key: "name",
  value: "",
  link: "",
  symbol: "",
};
